server/session/handler_inventory_transaction.go

```diff
@@ -141,6 +141,7 @@ func (h *InventoryTransactionHandler) handleUseItemTransaction(data *protocol.Us
 		s.c.BreakBlock(pos)
 	case protocol.UseItemActionClickBlock:
 		s.c.UseItemOnBlock(pos, cube.Face(data.BlockFace), vec32To64(data.ClickedPosition))
+		s.c.UseItem()
```

Comment:
Are you sure this is a correct change? This sounds like a change that has quite some implications.

Reply:
Tal recommened I make that change. This was added in order to fix every raw food type that interacted with the campfire from instantly getting placed. The ideal behavior here is for each click to add a signular raw food type onto the campfire. I'm not really sure how to attain this behaviour.

Reply:
@JustTalDevelops could you elaborate on this? I'm not entirely sure what the issue is here.

Reply:
any news?

Reply:
> The ideal behavior here is for each click to add a signular raw food type onto the campfire. I'm not really sure how to attain this behaviour.

You need to send campfire and soul campfire recipes in CraftingDataPacket to achieve this behaviour.